packages/react-cosmos-redux-proxy/src/index.js

```diff
@@ -31,14 +28,44 @@ export default function createReduxProxy(options) {
       };
     }
 
+    componentWillReceiveProps(nextProps) {
+      const oldReduxState = this.props.fixture[fixtureKey];
+      const newReduxState = nextProps.fixture[fixtureKey];
+      if (oldReduxState !== newReduxState) {
+        this.reloadStore(nextProps);
+      }
+    }
+
     componentWillMount() {
```

Comment:
I know this is not your change, but can you change this `componentDidMount` because it's a better place for side effects? 

Side note: Afaik `componentWillMount` is considered harmful nowdays and doesn't have any proper use case.